### PR TITLE
rpc: Improve RPC help by explicitly mentioning output types

### PR DESCRIPTION
**Pull request by kiminuo**
This PR attempts to replicate https://github.com/bitcoin/bitcoin/blob/0ccf9b2e5594581deef2f60174c3651a57f93b64/src/rpc/rawtransaction.cpp#L547 to one other place (at the moment) so that users have better idea what RPC methods can actually return.

I created this PR as a follow-up to the idea mentioned here https://github.com/bitcoin/bitcoin/pull/23320#discussion_r732458112 (resolved).

### CODE REVIEW

**Comment by kiminuo** [comment]
cc @laanwj given he said https://github.com/bitcoin/bitcoin/pull/23320#discussion_r755083186.

**Comment by laanwj** [comment]
Concept ACK

**Comment by michaelfolkson** [comment]
Concept ACK

Seems like an improvement to display all the output types in the RPC help rather than just the `pubkeyhash` example.

**Comment by kiminuo** [comment]
Does it make sense to you to use `GetAllOutputTypes` where possible. I mean in RPC help sections?

Is the output OK? Or are possibly any output types that you would NOT show?

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1037,7 +1037,7 @@ static RPCHelpMan getblock()
                                             {RPCResult::Type::STR, "asm", "The asm"},
                                             {RPCResult::Type::STR, "hex", "The hex"},
                                             {RPCResult::Type::STR, "address", /* optional */ true, "The Bitcoin address (only if a well-defined address exists)"},
-                                            {RPCResult::Type::STR, "type", "The type, eg 'pubkeyhash'"},
+                                            {RPCResult::Type::STR, "type", "The type (" + GetAllOutputTypes() + ")"},
```
Comment (luke-jr) [inline src/rpc/blockchain.cpp:1040]: nit

Existing other case says "(e.g. " ...

Maybe "(one of: " would be better.

**Review by luke-jr — APPROVED** [review]
utACK 78d9227ced0442c10249a08ba86386f7b3d2755a

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (prusnak) [reply src/rpc/blockchain.cpp:1040]: I agree that `one of:` sounds good if we are enumerating all possible options.

Let's also change `src/rpc/rawtransaction.cpp:568`

**Comment by theStack** [comment]
Concept ACK

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (kiminuo) [reply src/rpc/blockchain.cpp]: Thank you. Modified.

**Review by kristapsk — APPROVED** [review]
ACK c9ca3e8bab76da8ab20c362c8cd541443e2e8d77

**Review by prusnak — APPROVED** [review]
Approach ACK c9ca3e8bab76da8ab20c362c8cd541443e2e8d77

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -574,7 +564,7 @@ static RPCHelpMan decodescript()
                  {
                      {RPCResult::Type::STR, "asm", "String representation of the script public key"},
                      {RPCResult::Type::STR_HEX, "hex", "Hex string of the script public key"},
-                     {RPCResult::Type::STR, "type", "The type of the script public key (e.g. witness_v0_keyhash or witness_v0_scripthash)"},
+                     {RPCResult::Type::STR, "type", "The type of the script public key (one of: " + GetAllOutputTypes() + ")"},
```
Comment (MarcoFalke) [inline src/rpc/rawtransaction.cpp:567]: pretty sure this is wrong. Only v0 can be wrapped, no?

Comment (kiminuo) [reply src/rpc/rawtransaction.cpp]: Reverted. Thanks.

**Review by kristapsk — APPROVED** [review]
re-ACK c821ab8be8dffb749853c05e05cb515c11e6328a

**Pull request merged**